noxfile.py

```diff
@@ -0,0 +1,247 @@
+
+import nox
+
+try:
+    import pytest
+
+    NO_TESTS_COLLECTED_EXIT_CODE = pytest.ExitCode.NO_TESTS_COLLECTED
+except ImportError:
+    NO_TESTS_COLLECTED_EXIT_CODE = 5
+
+# -- nox configuration --
+nox.options.default_venv_backend = "uv"
+nox.options.sessions = [
+    "test_cartesian-3.10(internal, cpu)",
+    "test_cartesian-3.10(dace, cpu)",
+    "test_cartesian-3.11(internal, cpu)",
+    "test_cartesian-3.11(dace, cpu)",
+    "test_eve-3.10",
+    "test_eve-3.11",
+    "test_next-3.10(internal, cpu, nomesh)",
+    "test_next-3.10(dace, cpu, nomesh)",
+    "test_next-3.11(internal, cpu, nomesh)",
+    "test_next-3.11(dace, cpu, nomesh)",
+    "test_storage-3.10(cpu)",
+    "test_storage-3.11(cpu)",
+]
+
+# -- Parameter sets --
+DeviceOption: TypeAlias = Literal["cpu", "cuda11", "cuda12", "rocm4_3", "rocm5_0"]
+DeviceNoxParam: Final = types.SimpleNamespace(
+    **{device: nox.param(device, id=device, tags=[device]) for device in DeviceOption.__args__}
+)
+DeviceTestSettings: Final[dict[str, dict[str, Sequence]]] = {
+    "cpu": {"extras": [], "markers": ["not requires_gpu"]},
+    **{
+        device: {"extras": [device], "markers": ["requires_gpu"]}
+        for device in ["cuda11", "cuda12", "rocm4_3", "rocm5_0"]
+    },
+}
+
+CodeGenOption: TypeAlias = Literal["internal", "dace"]
+CodeGenNoxParam: Final = types.SimpleNamespace(
+    **{
+        codegen: nox.param(codegen, id=codegen, tags=[codegen])
+        for codegen in CodeGenOption.__args__
+    }
+)
+CodeGenTestSettings: Final[dict[str, dict[str, Sequence]]] = {
+    "internal": {"extras": [], "markers": ["not requires_dace"]},
+    "dace": {"extras": ["dace"], "markers": ["requires_dace"]},
+}
+
+
+# -- nox sessions --
+@nox.session(python=["3.10", "3.11"], tags=["cartesian"])
+@nox.parametrize("device", [DeviceNoxParam.cpu, DeviceNoxParam.cuda12])
+@nox.parametrize("codegen", [CodeGenNoxParam.internal, CodeGenNoxParam.dace])
+def test_cartesian(
+    session: nox.Session,
+    codegen: CodeGenOption,
+    device: DeviceOption,
+) -> None:
+    """Run selected 'gt4py.cartesian' tests."""
+
+    codegen_settings = CodeGenTestSettings[codegen]
+    device_settings = DeviceTestSettings[device]
+
+    _install_session_venv(
+        session,
+        extras=["performance", "testing", *codegen_settings["extras"], *device_settings["extras"]],
+        groups=["test"],
+    )
+
+    num_processes = session.env.get("NUM_PROCESSES", "auto")
+    markers = " and ".join(codegen_settings["markers"] + device_settings["markers"])
+
+    session.run(
+        *f"pytest --cache-clear -sv -n {num_processes}".split(),
+        *("-m", f"{markers}"),
+        str(pathlib.Path("tests") / "cartesian_tests"),
+        *session.posargs,
+    )
+    session.run(
+        *"pytest --doctest-modules --doctest-ignore-import-errors -sv".split(),
+        str(pathlib.Path("src") / "gt4py" / "cartesian"),
+    )
+
+
+@nox.session(python=["3.10", "3.11"])
+def test_docs(session: nox.Session) -> None:
```

Comment:
I would not call it docs, because I connect `docs` with sphinx documentation, more than with executable notebooks.

Reply:
Renamed to `test_examples`